spec/lib/govuk_proxy/static_proxy_spec.rb

```diff
@@ -0,0 +1,40 @@
+require "spec_helper"
+require "govuk_app_config/govuk_proxy/static_proxy"
+require "rack/mock"
+
+RSpec.describe GovukProxy::StaticProxy do
+  # dummy app to validate success
+  let(:app) { ->(_env) { [200, {}, "success"] } }
+  let(:app_domain) { "app.domain" }
+  let(:static_domain) { "static.domain" }
+
+  subject { GovukProxy::StaticProxy.new(app, backend: "https://static.domain", streaming: false) }
+
+  it "redirects the request if path begins with /asset/static" do
+    test_request("/assets/static/a.css", true)
+  end
+
+  it "ignores requests not with path prefix /asset/static" do
+    test_request("/assets/app/a.css", false)
+  end
+
+  it "ignores requests where /asset/static isn't a prefix" do
+    test_request("/another/prefix/assets/static/a.css", false)
+  end
+
+  it "ignores requests with no path" do
+    test_request("/", false)
+  end
+end
+
+def test_request(path, proxied)
```

Comment:
You should define this helper method within the RSpec describe method, as it'd be only scoped to tests there, whereas here it is a global function available to the whole app (defined as a method on `Object`)

Reply:
moved